home/views.py

```diff
@@ -1,81 +1,208 @@
+from __future__ import unicode_literals
 from django.shortcuts import render, redirect
-from django.http import HttpResponse
 from django.contrib.auth import logout
+from django.contrib.auth.models import Group
+from django.contrib.auth.models import User as DjangoUser
 from django.contrib.auth.decorators import login_required
-from django.contrib.auth.forms import UserCreationForm
-from .forms import UserRegisterForm
-from .models import Users
+from .forms import UserRegisterForm, ChangePwdForm
 from django.contrib import messages
+from .models import Contact, User, Individual, ClassOffering, ClassEnrollment
+from django.contrib.auth import update_session_auth_hash
 
+
+'''
+**THIS index() method WILL NOT BE USED IN PRODUCTION: **
+it is a method that was added to test the mb-salesforce database 
+connection during development
+This method should be used to test our database, we can push data to it through here via a
+post request and we can query the database from here as well.  Leave this method here for
+ease of fxnality testing purposes
+until we are ready to deploy in production. 
+'''
+
+
+def index(request):
+    contacts = Contact.objects.all()
+    users = User.objects.all()
+    individuals = Individual.objects.all()
+    classes = ClassOffering.objects.all()
+    enrollments = ClassEnrollment.objects.all()
+    context = {
+        'contacts': contacts,
+        'users' : users,
+        'individuals' : individuals,
+        'classes' : classes,
+        'enrollments' : enrollments
+    }
+    return render(request, 'home/index.html', context)
+
+
+'''
+If the request's user already has a tag they are redirected to the correct page.  When a user
+first logs in they don't have a tag and they are directed based on the group they belong to with
+approximately the highest permission level.  If the user wants to use the app as a member of
+a different group, they can change in their profile, their tag will be changed and this method 
+will be called with a reqeust.user with a tag.
+redirects ref:
+https://realpython.com/django-redirects/#django-redirects-a-super-simple-example
+'''
 @login_required
 def home(request):
-    user = Users.objects.filter(email = str(request.user.email)).first()
-    if user == None:
-        # User should only be able to get here if they logged in with google AND they haven't registered with MB yet
-        return redirect('home-register_after_oauth')
-    elif user.role == 'staff':
-        return redirect('staff')
-    elif user.role == 'student':
-        return redirect('student')
-    elif user.role == 'teacher':
-        return redirect('teacher')
-    else: #user.role == 'volunteer'
-        return redirect('volunteer')
+    if request.user.userprofile.change_pwd:
```

Comment:
I think we might have to change how we access `userprofile` here as when I did a local login with google I got an exception:

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/home/

Django Version: 2.2.1
Python Version: 3.7.3
Installed Applications:
['student.apps.StudentConfig',
 'teacher.apps.TeacherConfig',
 'volunteer.apps.VolunteerConfig',
 'staff.apps.StaffConfig',
 'donor.apps.DonorConfig',
 'home.apps.HomeConfig',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'social_django',
 'crispy_forms',
 'coverage',
 'salesforce']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback:

File "/Users/bob/src/MB_Portal/venv/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/Users/bob/src/MB_Portal/venv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/Users/bob/src/MB_Portal/venv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/Users/bob/src/MB_Portal/venv/lib/python3.7/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  21.                 return view_func(request, *args, **kwargs)

File "/Users/bob/src/MB_Portal/home/views.py" in home
  51.     if request.user.userprofile.change_pwd:
```

Reply:
This could also be caused by the migration to add the userprofile not being complete, if the code otherwise guarantees that this companion row always exists now.

Reply:
Interesting, as soon as you log in you should be getting a UserProfile and a User model row auto-generated (below), and I'm unable to replicate.  It could be a migration issue, we'll see if it comes up in the future:

```python
class UserProfile(mdls.Model):
    user = mdls.OneToOneField(DjangoUser, on_delete=mdls.CASCADE)
    change_pwd = mdls.BooleanField(default=False)
    date_of_birth = mdls.DateField(default="1901-01-01")
    salesforce_id = mdls.CharField(default="xxxxxx01011901", max_length=14)

    @receiver(post_save, sender=DjangoUser)
    def create_user_profile(sender, instance, created, **kwargs):
        if created:
            UserProfile.objects.create(user=instance)

    @receiver(post_save, sender=DjangoUser)
    def save_user_profile(sender, instance, **kwargs):
        instance.userprofile.save()
```